src/lib/statistics/statistics_objects/min_max_filter.cpp

```diff
@@ -158,6 +159,52 @@ bool MinMaxFilter<T>::does_not_contain(const PredicateCondition predicate_condit
       const auto value2 = boost::get<T>(*variant_value2);
       return value >= max || value2 <= min;
     }
+    case PredicateCondition::Like: {
+      if constexpr (std::is_same_v<T, pmr_string>) {
+        const auto pattern = boost::get<pmr_string>(variant_value);
+
+        if (!LikeMatcher::contains_wildcard(pattern)) {
+          return value < min || value > max;
+        }
+
+        if (LikeMatcher::get_index_of_next_wildcard(pattern, 0) == 0) {
+          return false;
+        }
+
+        const auto [lower_bound, upper_bound] = LikeMatcher::get_lower_upper_bound(pattern);
+
```

Comment:
It now looks as if only part of the code is moved into the `LikeMatcher` (see above)